### PR TITLE
Add ipm_image_node

### PR DESCRIPTION
**Pull request by Flova**
Move mask and image projections from `soccer_ipm` to own node as discussed in the soccer_ipm PR.

### CODE REVIEW

**Comment by ijnek** [comment]
Thanks for this PR @Flova! Looks good overall.

I think it would be good to name the package ipm_image rather than image_ipm for consistency (and to help tab-completion and ipm_* wildcards)

File: image_ipm/image_ipm/ipm.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        # Masks
+        mask_publisher = self.create_publisher(PointCloud2, 'mask_relative_pc', 1)
+
+        self.create_subscription(
+            Image,
+            'mask_in_image',
+            lambda msg: mask_publisher.publish(self.map_mask(msg)),
+            1)
+
+    def map_mask(self, msg: Image) -> PointCloud2:
```
Comment (jaagut) [inline image_ipm/image_ipm/ipm.py]: As this node also supports colored images, this should be reflected in the naming scheme.

File: image_ipm/image_ipm/ipm.py [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+        # We need to create a tf buffer
+        self.tf_buffer = tf2.Buffer(cache_time=Duration(seconds=30.0))
+        self.tf_listener = tf2.TransformListener(self.tf_buffer, self)
+
+        # Create an IPM instance
+        self.ipm = IPM(self.tf_buffer)
+
+        # Declare params
+        self.declare_parameter('output_frame', 'base_footprint')
+        self.declare_parameter('type', 'mask')
```
Comment (jaagut) [inline image_ipm/image_ipm/ipm.py]: What are other values, and how do they influence the behavior? This should be documented, e.g., in the README.
Perhaps also add some example images with results for cases `mask` and `rgb_image`.

File: image_ipm/test/test_ipm.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    tf.transform.rotation.w = 1.0
+
+    # Publish the dummy transform
+    tf_pub.publish(TFMessage(
+        transforms=[tf]
+    ))
+    # Spin the ipm to process the new data
+    rclpy.spin_once(node, timeout_sec=0.1)
+
+    node.set_parameters([Parameter("type", value=mode)])
```
Comment (jaagut) [inline image_ipm/test/test_ipm.py]:
```suggestion
    node.set_parameters([Parameter('type', value=mode)])
```
This throws off the CI.

File: image_ipm/image_ipm/ipm.py (hunk 2 above)
Comment (Flova) [reply image_ipm/image_ipm/ipm.py]: I agree, we should expand the docs later on with examples, etc..

**Comment by Flova** [comment]
> I think it would be good to name the package ipm_image rather than image_ipm for consistency (and to help tab-completion and ipm_* wildcards)

I renamed it to `ipm_image_node`.

**Review by ijnek — APPROVED** [review]
Thanks for applying the feedback!

**Pull request merged**